application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,417 @@
+            return jda.getTextChannelCache()
+                    .stream()
+                    .filter(channel -> fallbackChannelPattern.test(channel.getName()))
+                    .toList();
+        }
+    }
+
+    /**
+     * Provides the {@link EmbedBuilder} from an RSS item used for sending RSS posts.
+     *
+     * @param item the RSS item to construct the embed message from
+     * @param feedConfig the configuration of the RSS feed
+     * @return the constructed {@link EmbedBuilder} containing information from the RSS item
+     */
+    private static EmbedBuilder constructEmbedMessage(Item item, RSSFeed feedConfig) {
+        final EmbedBuilder embedBuilder = new EmbedBuilder();
+        String title = item.getTitle().orElse("No title");
+        String titleLink = item.getLink().orElse("");
+        Optional<String> rawDescription = item.getDescription();
+
+        // Set the item's timestamp to the embed if found
+        item.getPubDate()
+            .ifPresent(date -> embedBuilder
+                .setTimestamp(getZonedDateTime(date, feedConfig.dateFormatterPattern())));
+
+        embedBuilder.setTitle(title, titleLink);
+        embedBuilder.setAuthor(item.getChannel().getLink());
+
+        // Process embed's description if a raw description was provided
+        if (rawDescription.isPresent() && !rawDescription.orElseThrow().isEmpty()) {
+            Document fullDescription =
+                    Jsoup.parse(StringEscapeUtils.unescapeHtml4(rawDescription.orElseThrow()));
+            String finalDescription = fullDescription.body()
+                .select("*")
+                .stream()
+                .map(Element::text)
+                .collect(Collectors.joining(". "));
+
+            embedBuilder.setDescription(StringUtils.abbreviate(finalDescription, MAX_CONTENTS));
+        } else {
+            embedBuilder.setDescription("No description");
+        }
+
+        return embedBuilder;
+    }
+
+    /**
+     * Fetches a list of {@link Item} from a given RSS url.
+     *
+     * @param rssUrl the URL of the RSS feed to fetch
+     * @return a list of {@link Item} parsed from the RSS feed, or an empty list if there's an
+     *         {@link IOException}
+     */
+    private List<Item> fetchRSSItemsFromURL(String rssUrl) {
+        try {
+            return rssReader.read(rssUrl).toList();
+        } catch (IOException e) {
+            logger.warn("Could not fetch RSS from URL ({})", rssUrl);
+            return List.of();
+        }
+    }
+
+    /**
+     * Helper function for parsing a given date value to a {@link ZonedDateTime} with a given
+     * format.
+     *
+     * @param date the date value to parse, can be null
+     * @param format the format pattern to use for parsing
+     * @return the parsed {@link ZonedDateTime} object
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private static ZonedDateTime getZonedDateTime(@Nullable String date, @NotNull String format)
```

Comment:
No need to annotate with @NotNull if you added it in the `package-info.java` WHICH is btw missing, you should add it.

Reply:
Done!